packages/@aws-cdk/aws-lambda-golang/lib/bundling.ts

```diff
@@ -0,0 +1,162 @@
+import * as os from 'os';
+import * as path from 'path';
+import { AssetCode, Code, Runtime } from '@aws-cdk/aws-lambda';
+import * as cdk from '@aws-cdk/core';
+import { BundlingOptions } from './types';
+import { exec, findUp, getGoBuildVersion } from './util';
+
+/**
+ * Options for bundling
+ */
+export interface BundlingProps extends BundlingOptions {
+  /**
+   * Path to go.mod file
+   */
+  readonly modFilePath: string;
+
+  /**
+   * Entry path
+   */
+  readonly entry: string;
+
+  /**
+   * The runtime of the lambda function
+   */
+  readonly runtime: Runtime;
+}
+
+/**
+ * Bundling
+ */
+export class Bundling implements cdk.BundlingOptions {
+
+  public static bundle(options: BundlingProps): AssetCode {
+    return Code.fromAsset(path.dirname(options.modFilePath), {
+      assetHashType: options.assetHashType ?? cdk.AssetHashType.OUTPUT,
+      assetHash: options.assetHash,
+      bundling: new Bundling(options),
```

Comment:
Related to https://github.com/aws/aws-cdk/pull/11842#issuecomment-823154126

Maybe the solution is to `delete props` before passing this to `bundling`.

Or explicitly only pass the `public` properties...